Test failure when building on Fedora rawhide (36): AssertionError: "^can't\ set\ attribute$" does not match "can't set attribute 'key_attr'"

## Description

When building hdmf 3.1.1 on Fedora, one test fails with:

```
======================================================================
FAIL: test_set_key_attr (utils_test.test_labelleddict.TestLabelledDict)
Test that the key attribute cannot be set after initialization.
----------------------------------------------------------------------
AttributeError: can't set attribute 'key_attr'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/builddir/build/BUILD/hdmf-3.1.1/tests/unit/utils_test/test_labelleddict.py", line 36, in test_set_key_attr
    with self.assertRaisesWith(AttributeError, "can't set attribute"):
AssertionError: "^can't\ set\ attribute$" does not match "can't set attribute 'key_attr'"

----------------------------------------------------------------------
Ran 1220 tests in 7.455s

FAILED (failures=1, skipped=8, expected failures=1)
======================================================================
2021-08-16 18:55:36,332 - INFO - Ran 1220 tests - FAILED (failures=1)

```
The final regex isn't matching: should it be using `assertRaisesWith` here, which I see does an exact match?

## Steps to Reproduce

<!--Provide a minimal code snippet here to reproduce this error.-->

N/A

## Environment

<!--Please describe your environment according to the following bullet points.-->

    Python Executable: python
    Python Version: Python 3.7, 3.8, or 3.9: python3.10
    Operating System: Windows, macOS or Linux: Fedora Linux
    HDMF Version: 3.1.1

## Checklist

- [x] Have you ensured the bug was not already [reported](https://github.com/hdmf-dev/hdmf/issues) ?
- [x] Have you included a brief and descriptive title?
- [x] Have you included a clear description of the problem you are trying to solve?
- [-] Have you included a minimal code snippet that reproduces the issue you are encountering?
- [-] Have you checked our [Contributing](https://github.com/hdmf-dev/hdmf/blob/dev/docs/CONTRIBUTING.rst) document?

Build logs attached:

[hdmf-buildlog.txt](https://github.com/hdmf-dev/hdmf/files/6994432/hdmf-buildlog.txt)
[hdmf-rootlog.txt](https://github.com/hdmf-dev/hdmf/files/6994433/hdmf-rootlog.txt)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,8 +1,13 @@
 # HDMF Changelog
 
+## HDMF 3.1.2 (TBD)
+
+### Bug fixes
+- Fix broken tests in Python 3.10. @rly (#664)
+
 ## HDMF 3.1.1 (July 29, 2021)
 
-### Fixes
+### Bug fixes
 - Updated the new ``DynamicTableRegion.get_linked_tables`` function (added in 3.1.0) to return lists of ``typing.NamedTuple``
   objects rather than lists of dicts. @oruebel (#660)
 

--- a/tests/unit/utils_test/test_labelleddict.py
+++ b/tests/unit/utils_test/test_labelleddict.py
@@ -33,7 +33,7 @@ class TestLabelledDict(TestCase):
     def test_set_key_attr(self):
         """Test that the key attribute cannot be set after initialization."""
         ld = LabelledDict(label='all_objects')
-        with self.assertRaisesWith(AttributeError, "can't set attribute"):
+        with self.assertRaises(AttributeError):
             ld.key_attr = 'another_name'
 
     def test_getitem_unknown_val(self):

